### PR TITLE
cmd: pull closure out into its own func and add config.go

### PR DESCRIPTION
**Pull request by jzelinskie**
...wow the diff sucks for this. I basically just pulled the closure out into its own function that returns an error and then moved the config stuff into its own file.

### CODE REVIEW

File: cmd/chihaya/main.go [hunk 1]
```diff
@@ -12,178 +11,138 @@ import (
...
+		pprof.StartCPUProfile(f)
+		defer pprof.StopCPUProfile()
 	}
 
-	f, err := os.Open(os.ExpandEnv(path))
+	configFilePath, _ := cmd.Flags().GetString("config")
+	configFile, err := ParseConfigFile(configFilePath)
 	if err != nil {
-		return nil, err
+		return errors.New("failed to read config: " + err.Error())
```
Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: We could start using github.com/pkg/errors and make this to `errors.Wrap(err,"failed to read config")`. I use the package for pretty much every new project I start because I have to wrap so often.

File: cmd/chihaya/main.go [hunk 2]
```diff
@@ -12,178 +11,138 @@ import (
...
+			if err := rootCmdRun(cmd, args); err != nil {
 				log.Fatal(err)
 			}
 		},
 	}
-
-	rootCmd.Flags().StringVar(&configFilePath, "config", "/etc/chihaya.yaml", "location of configuration file")
-	rootCmd.Flags().StringVarP(&cpuProfilePath, "cpuprofile", "", "", "location to save a CPU profile")
+	rootCmd.Flags().String("config", "/etc/chihaya.yaml", "location of configuration file")
+	rootCmd.Flags().StringP("cpuprofile", "", "", "location to save a CPU profile")
```
Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: I just noticed: You made this a string parameter with `""` as the shorthand - intended? Why?
Also, let's give `config` a shorthand, now that I know about the feature :)

File: cmd/chihaya/main.go (hunk 1 above)
Comment (jzelinskie) [inline cmd/chihaya/main.go]: I'd be ok with this. Maybe we should do it as a separate PR.

File: cmd/chihaya/main.go (hunk 2 above)
Comment (jzelinskie) [inline cmd/chihaya/main.go]: This was definitely a mistake. I'm not sure what good shorthand characters would be a good choice.

File: cmd/chihaya/main.go (hunk 1 above)
Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: Yup, sounds good

**Comment by mrd0ll4r** [comment]
LGTM
